src/auth/impersonated.ts

```
@@ -253,4 +255,18 @@ export class Impersonated extends OAuth2Client implements IdTokenProvider {
 
     return res.data.token;
   }
+
+  getTrustBoundaryUrl(): string {
+    const targetPrincipal = this.getTargetPrincipal();
+    if (!targetPrincipal) {
+      throw new Error(
```

Is this possible?

---

Yes, in the constructor we see:

`this.targetPrincipal = options.targetPrincipal ?? '';`

Which means targetPrincipal can have an empty string when the options don't have a corresponding value. This means that our tbUrl would be improperly configured

In which case, I think we should throw an error.